### DISCUSSION TITLE
Logging stack

### DISCUSSION THREAD

**Discussion by raphaelbernhart** [coollabsio/coolify#2229] [Q&A]
Hi there,

what Logging stack are you guys both using for logging the servers and more important the coolify apps - containers.

I know there is already support for Axiom and new Relic but are you guys using. I neither worked with Axiom nor new Relic so far.
Is anyone having a good setup with Grafana and Loki?

Thanks anyways :)

**Comment by carbontwelve** [comment]
I'm interested in using Loki as the log agregator with Grafana to display and filter. Have yet to find any details on how to do so with Coolify.

I can see fluentbit as one of the log sinks in Coolify and that supports [piping output to Loki](https://docs.fluentbit.io/manual/pipeline/outputs/loki). However, no documentation on uding fluentbit with Coolify.

**Reply by raphaelbernhart** [reply]
Yeah I came to the same conclusion, maybe I'll do some further research and share my findings :)

**Reply by bkovacev** [reply]
Have you guys figured this out :D ?

**Reply by raphaelbernhart** [reply]
I did not do further research tbh. I just used [Axiom](https://axiom.co/) as it satisfies my needs.

**Reply by lesimoes** [reply]
I did deploy granfa + loki with this docker compose: https://raw.githubusercontent.com/grafana/loki/v3.0.0/production/docker-compose.yaml

I run a simple test and work as spected.

Pay attention because this compose file runs a grafana dashboard without authentication, is highly recommended put your authentication on this file.

A little extra tip: If you are loggin your application is recommended use OpenTL to avoid vendo lock in, you could have more work to do but is better.

**Reply by ulisse1996** [reply]
If you want an all in one solution without a lot of work, for testing and dev you could use https://github.com/grafana/docker-otel-lgtm that have everything ready for logs, traces and metrics :)

**Reply by rubens-facilit** [reply]
> I did deploy granfa + loki with this docker compose: https://raw.githubusercontent.com/grafana/loki/v3.0.0/production/docker-compose.yaml
> 
> I run a simple test and work as spected.
> 
> Pay attention because this compose file runs a grafana dashboard without authentication, is highly recommended put your authentication on this file.
> 
> A little extra tip: If you are loggin your application is recommended use OpenTL to avoid vendo lock in, you could have more work to do but is better.

Hi, could you clarify what should I do after deploying this? Should I configure FluentBit in the server, or do I define something specific in each container?

**Reply by rubens-facilit** [reply]
> I'm interested in using Loki as the log agregator with Grafana to display and filter. Have yet to find any details on how to do so with Coolify.
> 
> I can see fluentbit as one of the log sinks in Coolify and that supports [piping output to Loki](https://docs.fluentbit.io/manual/pipeline/outputs/loki). However, no documentation on uding fluentbit with Coolify.

Hi, I figured it out and just posted a guide: https://github.com/coollabsio/coolify/discussions/5317
